### PR TITLE
Add is-integer() and is-decimal functions

### PR DESCRIPTION
**Pull request by ecerulm**

### CODE REVIEW

File: functions.md [hunk 1]
```diff
@@ -76,6 +76,33 @@ is-number(1.0)  => true
 is-number("1")  => false
 ```
 
+### _is-integer(object) -> boolean_
+
+True iff the argument is a integral number.
```
Comment (larsga) [inline functions.md]: "An" integral number. Sorry.

File: functions.md [hunk 2]
```diff
@@ -76,6 +76,33 @@ is-number(1.0)  => true
...
+
+Examples:
+
+```
+is-decimal(null) => false
+is-decimal(1)    => false
+is-decimal(1.0)  => true
+is-decimal("1.0")  => false
+```
+
```
Comment (larsga) [inline functions.md]: Thank you for adding documentation! Excellent! 💯

File: src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java [hunk 3]
```diff
@@ -760,6 +762,32 @@ public JsonNode call(JsonNode input, JsonNode[] arguments) {
...
+  // ===== IS-DECIMAL
+
+  public static class IsDecimal extends AbstractFunction {
+
+    public IsDecimal() {
+      super("is-decimal", 1, 1);
+    }
+
+    public JsonNode call(JsonNode input, JsonNode[] arguments) {
+      return NodeUtils.toJson(arguments[0].isDouble());
```
Comment (larsga) [inline src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java]: Is it possible that we might get an instance of `FloatNode` and that this method will return `false` for it?

https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/node/FloatNode.html

Looks like the answer is yes: https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/node/FloatNode.java

Could we add a test where we manually construct the input as a `FloatNode` using the Jackson API, and then attempt to call `is-decimal(.)`? `StaticTests` is the place for this.

Comment (ecerulm) [reply src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java]: yes, in fact. this wil fail for `FloatNode` and moreover `is-integer` will also fail for `BigDecimalNode`. I've changed this test to use `isIntegralNumber()` and `isFloatingPointNumber()`

**Pull request merged**